gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystemNewIntegrationTest.java

```diff
@@ -711,6 +711,42 @@ private void delete_directory(boolean parallelStatus) throws Exception {
     assertThat(gcsFs.exists(bucketUri.resolve(dirObject + "/d1"))).isFalse();
   }
 
+  @Test
+  public void delete_inferred_directory() throws Exception {
+    GoogleCloudStorageFileSystemOptions gcsFsOptions =
+        newGcsFsOptions().setStatusParallelEnabled(false).build();
+
+    TrackingHttpRequestInitializer gcsRequestsTracker =
+        new TrackingHttpRequestInitializer(httpRequestsInitializer);
+    GoogleCloudStorageFileSystem gcsFs = newGcsFs(gcsFsOptions, gcsRequestsTracker);
+
+    String bucketName = gcsfsIHelper.sharedBucketName1;
+    URI bucketUri = new URI("gs://" + bucketName + "/");
+    String dirObject = getTestResource();
+
+    gcsfsIHelper.createObjects(bucketName, dirObject + "/d1/d2");
```

Comment:
Use `f1` instead of `d2` because this is a file, not directory.

Reply:
Done